### ISSUE TITLE
CA2000 false positive when using DisposeWith

### ISSUE THREAD

**Issue by SuperJMN** [dotnet/roslyn-analyzers#7022]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Describe the bug

There are calls like this, part of [ReactiveUI](https://www.reactiveui.net) that are perfectly OK, but the warning is triggered.

`myDisposable.DisposeWith(compositeDisposable);`

Internally, the call does something like this:

`compositeDisposable.Add(myDisposable);`

### Steps To Reproduce

1. Create any kind of C# application. Console application works, too.
2. Install the [ReactiveUI](https://www.nuget.org/packages/ReactiveUI/)
3. Write something like this

```csharp
CompositeDisposable disposable = new();

new MyClass().DiposeWith(disposable);

public class MyClass : IDisposable
{
	public void Dispose()
	{
	}
}
```

### Expected behavior

The warning shouldn't be triggered.

### Actual behavior

The warning IS triggered.
